protostar/testing/test_scheduler.py

```diff
@@ -2,16 +2,56 @@
 import signal
 from pathlib import Path
 from typing import TYPE_CHECKING, Callable, List, Optional
+import dataclasses
+
+from protostar.testing import (
+    TestingSummary,
+)
+from protostar.testing import (
+    BrokenTestSuiteResult,
+    TestResult,
+)
 
 from .test_collector import TestCollector
 from .test_runner import TestRunner
 from .test_shared_tests_state import SharedTestsState
 from .testing_seed import Seed
 
+
 if TYPE_CHECKING:
     from protostar.commands.test.testing_live_logger import TestingLiveLogger
 
 
+def make_path_relative_if_possible(test_result: TestResult, path: Path) -> TestResult:
+    try:
+        test_result = dataclasses.replace(
+            test_result,
+            file_path=test_result.file_path.resolve().relative_to(path.resolve()),
+        )
+    except ValueError:
+        # We do this to preserve the functionality of running tests that are outside of the project
+        pass
+    return test_result
```

Comment:
not sure if this place is good for this function, alternatively I can move this to a separate file like `test_utils.py` or something. Bottom line is, there was a circular dependency error.